source/frontend/src/features/mapSideBar/property/tabs/propertyDetailsManagement/activity/detail/PropertyActivityDetailContainer.tsx

```diff
@@ -0,0 +1,125 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useOrganizationRepository } from '@/features/contacts/repositories/useOrganizationRepository';
+import { usePersonRepository } from '@/features/contacts/repositories/usePersonRepository';
+import { usePropertyActivityRepository } from '@/hooks/repositories/usePropertyActivityRepository';
+import {
+  Api_PropertyActivity,
+  Api_PropertyActivityInvolvedParty,
+  Api_PropertyMinistryContact,
+} from '@/models/api/PropertyActivity';
+
+import { IPropertyActivityDetailViewProps } from './PropertyActivityDetailView';
+
+export interface IPropertyActivityDetailContainerProps {
+  propertyId: number;
+  propertyActivityId: number;
+  onClose: () => void;
+  View: React.FunctionComponent<React.PropsWithChildren<IPropertyActivityDetailViewProps>>;
+}
+
+export const PropertyActivityDetailContainer: React.FunctionComponent<
+  React.PropsWithChildren<IPropertyActivityDetailContainerProps>
+> = ({ propertyId, propertyActivityId, onClose, View }) => {
+  const [show, setShow] = useState(true);
+
+  const [loadedActivity, setLoadedActivity] = useState<Api_PropertyActivity | null>(null);
+
+  const {
+    getActivity: { execute: getActivity, loading: getActivityLoading },
+  } = usePropertyActivityRepository();
+
+  const {
+    getPersonDetail: { execute: getPerson, loading: loadingPerson },
+  } = usePersonRepository();
+
+  const {
+    getOrganizationDetail: { execute: getOrganization, loading: loadingOrganization },
+  } = useOrganizationRepository();
+
+  const fetchMinistryContacs = useCallback(
+    async (c: Api_PropertyMinistryContact) => {
+      if (c.personId !== undefined && c.personId !== null) {
+        const person = await getPerson(c.personId);
+        if (person !== undefined) {
+          c.person = person;
+        }
+      }
+    },
+    [getPerson],
+  );
+
+  const fetchPartiesContact = useCallback(
+    async (c: Api_PropertyActivityInvolvedParty) => {
+      if (c.personId !== undefined && c.personId !== null) {
+        const person = await getPerson(c.personId);
+        if (person !== undefined) {
+          c.person = person;
+        }
+      }
+      if (!!c.organizationId) {
```

Comment:
any reason why we are switching between explicitly checking for null and undefined above but then using a truthy check here?